internal/http/services/owncloud/ocdav/meta.go

```diff
@@ -59,3 +76,97 @@ func (h *MetaHandler) Handler(s *svc) http.Handler {
 
 	})
 }
+
+func (h *MetaHandler) handlePathForUser(w http.ResponseWriter, r *http.Request, s *svc, rid *provider.ResourceId) {
+	ctx, span := rtrace.Provider.Tracer("ocdav").Start(r.Context(), "meta_propfind")
+	defer span.End()
+
+	id := resourceid.OwnCloudResourceIDWrap(rid)
+	sublog := appctx.GetLogger(ctx).With().Str("path", r.URL.Path).Str("resourceid", id).Logger()
+	client, err := s.getClient()
+	if err != nil {
+		sublog.Error().Err(err).Msg("error getting grpc client")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	pf, status, err := propfind.ReadPropfind(r.Body)
+	if err != nil {
+		sublog.Debug().Err(err).Msg("error reading propfind request")
+		w.WriteHeader(status)
+		return
+	}
+
+	if ok := hasProp(&pf, net.PropOcMetaPathForUser); !ok {
+		sublog.Debug().Str("prop", net.PropOcMetaPathForUser).Msg("error finding prop in request")
+		w.WriteHeader(http.StatusBadRequest)
```

Comment:
@phil-davis tests are failing e.g. here https://drone.cernbox.cern.ch/cs3org/reva/6559/26/7 because I write a `400 bad request` when the `meta-path-for-user` is not in the request payload. That's the way I understood the docs (see https://doc.owncloud.com/server/next/developer_manual/webdav_api/meta.html#meta-files-xml ) regarding required props in the request payload. Do you have insights about this? Should the `meta-path-for-user` be returned even when not requested?! Or do the tests need to be adjusted when a propfind without the `meta-path-for-user` prop is being made? 🤔

Reply:
```
 Scenario: User cannot access meta folder of a file which is owned by somebody else        # /drone/src/tmp/testrunner/tests/acceptance/features/apiVersions/fileVersions.feature:124
Given user "Brian" has been created with default attributes and without skeleton files  # FeatureContext::userHasBeenCreatedWithDefaultAttributesAndWithoutSkeletonFiles()
And user "Alice" has uploaded file with content "123" to "/davtest.txt"                 # FeatureContext::userHasUploadedAFileWithContentTo()
And we save it into "FILEID"                                                            # FeatureContext::saveItInto()
When user "Brian" sends HTTP method "PROPFIND" to URL "/remote.php/dav/meta/<<FILEID>>" # FeatureContext::userSendsHTTPMethodToUrl()
Then the HTTP status code should be "404"                                               # FeatureContext::thenTheHTTPStatusCodeShouldBe()
HTTP status code 400 is not the expected value 404
Failed asserting that 400 matches expected '404'.
```

Reply:
in the tests the propfind is done without any body afaict.

Reply:
Nothing in the test scenarios has ever filled in the PROPFIND request body to explicitly ask for `oc:meta-path-for-user`

I created PR https://github.com/owncloud/core/pull/40010 to make the existing tests more flexible (allow both 400 or 404 response). That should help get this passing.

I will make a separate issue to make some tests that actually send the documented PROPFIND request body.

Reply:
PR #2780 has been merged to edge branch. @kulmann please rebase this PR and see what CI thinks.

Reply:
nice, thank you! rebased just now. let's see... https://drone.cernbox.cern.ch/cs3org/reva/6754

Reply:
@phil-davis I pushed a fix for invalid resource IDs.

NOTE: Reva has dropped support for base64 encoded fileIDs completely. So it makes no sense IMO to have them in the tests. Let us use the plain ones from your gherkin.